$NetBSD: patch-aa,v 1.5 2020/05/14 19:09:10 joerg Exp $

Remove the -pedantic flag from the configure script, it isn't terribly
useful for C++0x code.

Fix C++11 test case to actually work. The existing test fails with both
GCC and Clang.

--- configure.orig	2010-10-10 00:59:59.000000000 +0000
+++ configure
@@ -15328,7 +15328,6 @@ ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
     typedef check<int> check_type;
     check_type c;
-    check_type&& cr = c;
 int
 main ()
 {
@@ -15384,7 +15383,6 @@ ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
     typedef check<int> check_type;
     check_type c;
-    check_type&& cr = c;
 int
 main ()
 {
@@ -15441,7 +15439,6 @@ ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
     typedef check<int> check_type;
     check_type c;
-    check_type&& cr = c;
 int
 main ()
 {
@@ -17972,7 +17969,7 @@ if test "${ac_cv_safe_to_use_wconversion
   $as_echo_n "(cached) " >&6
 else
   save_CFLAGS="$CFLAGS"
-                     CFLAGS="-Wconversion -Werror -pedantic ${AM_CFLAGS} ${CFLAGS}"
+                     CFLAGS="-Wconversion -Werror ${AM_CFLAGS} ${CFLAGS}"
        cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -18011,7 +18008,7 @@ if test "${ac_cv_safe_to_use_Wconversion
   $as_echo_n "(cached) " >&6
 else
   save_CFLAGS="$CFLAGS"
-                           CFLAGS="-Wconversion -Werror -pedantic ${AM_CFLAGS} ${CFLAGS}"
+                           CFLAGS="-Wconversion -Werror ${AM_CFLAGS} ${CFLAGS}"
          cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -18071,7 +18068,7 @@ if test "${ac_cv_safe_to_use_Wextra_+set
   $as_echo_n "(cached) " >&6
 else
   save_CFLAGS="$CFLAGS"
-         CFLAGS="${W_FAIL} -pedantic -Wextra ${AM_CFLAGS} ${CFLAGS}"
+         CFLAGS="${W_FAIL} -Wextra ${AM_CFLAGS} ${CFLAGS}"
          cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -18098,7 +18095,7 @@ fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_safe_to_use_Wextra_" >&5
 $as_echo "$ac_cv_safe_to_use_Wextra_" >&6; }
 
-      BASE_WARNINGS="${W_FAIL} -pedantic -Wall -Wundef -Wshadow ${NO_UNUSED} ${F_DIAGNOSTICS_SHOW_OPTION} ${CFLAG_VISIBILITY} ${BASE_WARNINGS_FULL}"
+      BASE_WARNINGS="${W_FAIL} -Wall -Wundef -Wshadow ${NO_UNUSED} ${F_DIAGNOSTICS_SHOW_OPTION} ${CFLAG_VISIBILITY} ${BASE_WARNINGS_FULL}"
       if test "$ac_cv_safe_to_use_Wextra_" = "yes"; then :
   BASE_WARNINGS="${BASE_WARNINGS} -Wextra"
 else
@@ -18120,7 +18117,7 @@ ac_link='$CXX -o conftest$ac_exeext $CXX
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
          save_CXXFLAGS="$CXXFLAGS"
-         CXXFLAGS="-Werror -pedantic -Wmissing-declarations ${AM_CXXFLAGS}"
+         CXXFLAGS="-Werror -Wmissing-declarations ${AM_CXXFLAGS}"
          cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -18163,7 +18160,7 @@ if test "${ac_cv_safe_to_use_Wlogical_op
   $as_echo_n "(cached) " >&6
 else
   save_CFLAGS="$CFLAGS"
-         CFLAGS="${W_FAIL} -pedantic -Wlogical-op ${AM_CFLAGS} ${CFLAGS}"
+         CFLAGS="${W_FAIL} -Wlogical-op ${AM_CFLAGS} ${CFLAGS}"
          cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -18205,7 +18202,7 @@ ac_link='$CXX -o conftest$ac_exeext $CXX
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
          save_CXXFLAGS="${CXXFLAGS}"
-         CXXFLAGS="${W_FAIL} -pedantic -Wredundant-decls ${AM_CXXFLAGS}"
+         CXXFLAGS="${W_FAIL} -Wredundant-decls ${AM_CXXFLAGS}"
          cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -18255,7 +18252,7 @@ ac_link='$CXX -o conftest$ac_exeext $CXX
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
          save_CXXFLAGS="${CXXFLAGS}"
-         CXXFLAGS="${W_FAIL} -pedantic -Wattributes -fvisibility=hidden ${AM_CXXFLAGS}"
+         CXXFLAGS="${W_FAIL} -Wattributes -fvisibility=hidden ${AM_CXXFLAGS}"
          cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -18313,7 +18310,7 @@ if test "${ac_cv_safe_to_use_Wno_attribu
   $as_echo_n "(cached) " >&6
 else
   save_CFLAGS="$CFLAGS"
-         CFLAGS="${W_FAIL} -pedantic -Wno_attributes_ ${AM_CFLAGS} ${CFLAGS}"
+         CFLAGS="${W_FAIL} -Wno_attributes_ ${AM_CFLAGS} ${CFLAGS}"
          cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -21222,8 +21219,8 @@ ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
      if test -z "$ac_cv_cxx_cinttypes"; then
        cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-#include $ac_cv_cxx_cstdint;
-                       #include <$location>
+#include $ac_cv_cxx_cstdint
+#include <$location>
 int
 main ()
 {
